lib/elixir/test/elixir/keyword_test.exs

```diff
@@ -26,4 +26,22 @@ defmodule KeywordTest do
     assert Map.__info__(:functions) -- Keyword.__info__(:functions) ==
            [from_struct: 1]
   end
+
+  test "get_and_update/3" do
```

Comment:
We should have a better test name here: we're not testing everything about `get_and_update/3`, but just the raising behaviour. Maybe `test "get_and_update/3 raises on bad return value from the argument function"` or something like this (same goes for the other two tests).